identity/README.md

```diff
@@ -0,0 +1,279 @@
+
+There might be operations that need to run on the CSI Controller (provisioner)
+for storage system access. Other operations may require running on a volume
+that is attached/staged on a node. By specifying the type of Service that the
+CSI-driver provides, the CSI-Addons CO plugin can make informed decisions on
+where to invoke the RPCs for an operation.
+
+```protobuf
+// GetCapabilitiesRequest is sent by the CSI-Addons CO plugin to detect the
+// features that a CSI-driver supports.
+message GetCapabilitiesRequest {
+  // Intentionally empty.
+}
+
+// GetCapabilitiesResponse is returned by the CSI-driver as a response to a
+// GetCapabilitiesRequest.
+message GetCapabilitiesResponse {
+  // All the capabilities that the controller service supports. This
+  // field is OPTIONAL.
+  repeated Capability capabilities = 1;
+}
+
+// Specifies one or more capabilities of the CSI-driver.
+message Capability {
+  // Service contains the type of CSI Service that the CSI-driver provides.
+  message Service {
+    // Type describes a CSI Service that CSI-drivers can support.
+    enum Type {
+      // UNKNOWN indicates that the CSI-driver does not neither provide the CSI
+      // ControllerService or CSI NodeService. The CSI-Addons CO plugin will
+      // most likely ignore the node providing this Identify Service.
+      UNKNOWN = 0;
+
+      // CONTROLLER_SERVICE indicates that the CSI-driver provides RPCs for the
+      // CSI ControllerService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require access to the storage system,
+      // similar to the CSI Controller (provisioner).
+      CONTROLLER_SERVICE = 1;
+
+      // NODE_SERVICE indicates that the CSI-driver provides RPCs for the CSI
+      // NodeService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require a volume to be staged/attached to
+      // the node.
+      NODE_SERVICE = 2;
+    }
+    // type contains the Type of CSI Service that the CSI-driver supports.
+    Type type = 1;
+  }
+
+  // Additional CSI-Addons operations will need to be added here.
+
+  oneof type {
+    // Service or operation that the CSI-driver supports.
+    Service service = 1;
+
+    // Additional CSI-Addons operations need to be appended to this list.
+  }
+}
+```
+
+#### GetCapabilities Errors
+
+If the CSI-driver is unable to complete the GetCapabilities call successfully,
+it MUST return a non-ok gRPC code in the gRPC status.
+
+### `Probe`
+
+The primary utility of the Probe RPC is to verify that the CSI-driver is in a
+healthy and ready state. If an unhealthy state is reported, via a non-success
+response, a CSI-Addons CO plugin MAY take action with the intent to bring the
+CSI-driver to a healthy state. Such actions MAY include, but SHALL NOT be
+limited to, the following:
+
+* Restarting the CSI-driver container, or
+* Notifying the CSI-driver supervisor.
```

Comment:
ooc, what is meant by "CSI driver supervisor" here? is it meant that an event or similiar will be generated?

Reply:
That would be the CO or maybe an operator for the CSI-driver. Reporting an event is one option.